app/lib/searching.rb

```diff
@@ -0,0 +1,62 @@
+class Searching
+  require 'gds_api/rummager'
+  attr_reader :params
+  def initialize(params)
+    @params = params
+  end
+
+  def count
+    return 10 if params["count"].blank?
+    return 1000 if params["count"].to_i > 1000
+    params["count"]
+  end
+
+  def call
+    terms = params["search_term"].split.join("+")
+    rummager = GdsApi::Rummager.new(Plek.new.find('rummager'))
+    fields = %w(
+      title link public_timestamp description format
+      )
+    findings_new_left = rummager.search(q: terms, fields: fields, count: count.to_s,
+      ab_tests: "#{params['which_test']}:A", c: Time.now.getutc.to_s)
+    findings_new_right = rummager.search(q: terms, fields: fields, count: count.to_s,
+      ab_tests: "#{params['which_test']}:B", c: Time.now.getutc.to_s)
+    Results.new(findings_new_left, findings_new_right)
+  end
+
+  class Results
+    attr_reader :left_total, :left_missing, :right_total, :right_missing, :left, :right, :result_count
+    def initialize(left, right)
+      @result_count = right['results'].count > left['results'].count ? right['results'].count : left['results'].count
+      @left = (0..result_count - 1).map { |i| Result.new(left['results'][i]) }
+      @right = (0..result_count - 1).map { |i| Result.new(right['results'][i]) }
+      @left_total = left['total']
+      @right_total = right['total']
+      @left_missing = @left_total - left['results'].count
+      @right_missing = @right_total - right['results'].count
+    end
+
+    def rows
+      left.zip(right)
+    end
+
+    # => tests
+    def search_right_list_for_link(link)
+      right.each { |result| return right.index(result) if result.info['link'].include?(link) }
+    end
+
+    def score_difference(link, position)
+      return search_right_list_for_link(link) - position if search_right_list_for_link(link)
+      "++++"
```

Comment:
yeah, I had just meant for this to mean "out of search" but that does make a lot of sense. Maybe I could just return 1001 seeing as this is more than could ever be returned it would make more sense?